rdmo/accounts/socialaccount.py

```diff
@@ -0,0 +1,45 @@
+from django.conf import settings
+from django.contrib.auth.models import Group
+from django.forms import BooleanField
+
+from allauth.socialaccount.adapter import DefaultSocialAccountAdapter
+from allauth.socialaccount.forms import SignupForm as AllauthSocialSignupForm
+
+from rdmo.accounts.forms import ProfileForm
+from rdmo.accounts.models import ConsentFieldValue
+
+
+class SocialAccountAdapter(DefaultSocialAccountAdapter):
+
+    def is_open_for_signup(self, request, sociallogin):
+        return settings.SOCIALACCOUNT_SIGNUP
+
+    def save_user(self, request, sociallogin, form=None):
+        user = super().save_user(request, sociallogin, form)
+
+        if settings.SOCIALACCOUNT_GROUPS:
+            provider = str(sociallogin.account.provider)
+            groups = Group.objects.filter(name__in=settings.SOCIALACCOUNT_GROUPS.get(provider, []))
+            user.groups.set(groups)
+
+        return user
+
+
+class SocialSignupForm(AllauthSocialSignupForm, ProfileForm):
```

Comment:
this `SocialSignupForm`  is new which Ive added here so that it can be used in combination with the ToU setting